src/main/haskell/kore/src/Kore/Step/ExpandedPattern.hs

```diff
@@ -257,19 +277,20 @@ top =
 
 {-| 'isTop' checks whether an ExpandedPattern is equivalent to a top Pattern.
 -}
-isTop :: ExpandedPattern level variable -> Bool
```

Comment:
Why not renamining dropping `Term` from `TopBottomTerm` and its methods, and making `Predicated` instance of `TopBottom`?  `instance TopBottom term => instance TopBottom (Predicated level variable term) where ...` ?